syntax = "proto3";

package ragcrud_service;
option go_package = "github.com/milkyhoop/flow-executor/internal/proto;ragcrud_service";

import "google/protobuf/empty.proto";

// option go_package = "github.com/milkyhoop/template-service-golang/internal/delivery/pb/template;template";
// option python_package = "app.protos";

service RagCrudService {
  rpc DoSomething (Ragcrud_serviceRequest) returns (Ragcrud_serviceResponse);
  rpc HealthCheck (google.protobuf.Empty) returns (google.protobuf.Empty);

  rpc CreateRagDocument (CreateRagDocumentRequest) returns (RagDocumentResponse);
  rpc ListRagDocuments (ListRagDocumentsRequest) returns (ListRagDocumentsResponse);
  rpc GetRagDocument (GetRagDocumentRequest) returns (RagDocumentResponse);
  rpc UpdateRagDocument (UpdateRagDocumentRequest) returns (RagDocumentResponse);
  rpc UpdateRagDocumentBySearch (UpdateRagDocumentBySearchRequest) returns (RagDocumentResponse);
  rpc DeleteRagDocument (DeleteRagDocumentRequest) returns (RagDocumentResponse);
  rpc FuzzySearchDocuments (FuzzySearchRequest) returns (FuzzySearchResponse);
}

message Ragcrud_serviceRequest {
  string user_id = 1;
  string input = 2;
}

message Ragcrud_serviceResponse {
  string status = 1;
  string result = 2;
}

message CreateRagDocumentRequest {
  string tenant_id = 1;
  string title = 2;
  string content = 3;
  string source = 4;
  repeated string tags = 5;
}


message GetRagDocumentRequest {
  int32 id = 1;
}

message UpdateRagDocumentRequest {
  int32 id = 1;
  string title = 2;
  string content = 3;
}

message UpdateRagDocumentBySearchRequest {
  string tenant_id = 1;
  string search_content = 2;
  string new_content = 3;
}


message DeleteRagDocumentRequest {
  int32 id = 1;
}

message RagDocumentResponse {
  int32 id = 1;
  string title = 2;
  string content = 3;
}

message ListRagDocumentsRequest {
  string tenant_id = 1;
}

message ListRagDocumentsResponse {
  repeated RagDocumentResponse documents = 1;
}

message FuzzySearchRequest {
  string tenant_id = 1;
  string search_content = 2;
  float similarity_threshold = 3;
}

message FuzzySearchResponse {
  repeated RagDocumentResponse documents = 1;
}
